common4j/src/main/com/microsoft/identity/common/java/net/UrlConnectionHttpClient.java

```diff
@@ -0,0 +1,361 @@
+        return retryPolicy.attempt(new Callable<HttpResponse>() {
+            public HttpResponse call() throws IOException {
+                return executeHttpSend(request, new Consumer<HttpResponse>() {
+                    @Override
+                    public void accept(HttpResponse httpResponse) {
+                        recordHttpTelemetryEventEnd(httpResponse);
+                    }
+                });
+            }
+        });
+    }
+
+    private static HttpRequest constructHttpRequest(@NonNull HttpClient.HttpMethod httpMethod,
+                                                    @NonNull URL requestUrl,
+                                                    @NonNull Map<String, String> requestHeaders,
+                                                    byte[] requestContent) {
+
+        // Apply special backcompat behaviors for PATCH, if reqd
+        if (HttpClient.HttpMethod.PATCH == httpMethod) {
+            // Because HttpURLConnection predates RFC-5789, we need to fallback on POST w/ a backcompat
+            // workaround. See: https://stackoverflow.com/a/32503192/741827
+            httpMethod = HttpClient.HttpMethod.POST;
+            // This map may be immutable.
+            requestHeaders = new HashMap<>(requestHeaders);
+            requestHeaders.put("X-HTTP-Method-Override", HttpClient.HttpMethod.PATCH.name());
+        }
+
+        // Construct request
+        return new HttpRequest(
+                requestUrl,
+                requestHeaders,
+                httpMethod.name(), // HttpURLConnection doesn't natively support PATCH
+                requestContent,
+                null
+        );
+    }
+
+    /**
+     * Convert stream into the string.
+     *
+     * @param inputStream {@link InputStream} to be converted to be a string.
+     * @return The converted string
+     * @throws IOException Thrown when failing to access inputStream stream.
+     */
+    private String convertStreamToString(final InputStream inputStream) throws IOException {
+        try {
+            final BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+            final char[] buffer = new char[streamBufferSize];
+            final StringBuilder stringBuilder = new StringBuilder();
+            int charsRead;
+
+            while ((charsRead = reader.read(buffer)) > -1) {
+                stringBuilder.append(buffer, 0, charsRead);
+            }
+
+            return stringBuilder.toString();
+        } finally {
+            safeCloseStream(inputStream);
+        }
+    }
+
+    /**
+     * Close the stream safely.
+     *
+     * @param stream stream to be closed
+     */
+    private static void safeCloseStream(final Closeable stream) {
+        if (stream == null) {
+            return;
+        }
+
+        try {
+            stream.close();
+        } catch (final IOException e) {
+            //Encountered IO exception when trying to close the stream"
```

Comment:
>//Encountered IO exception when trying to close the stream" [](http://example.com/codeflow?start=12&length=59)

Log Something, otherwise it will be harder to debug if we ever run in here. no?